nifi-nar-bundles/nifi-stateless-processor-bundle/nifi-stateless-processor/src/main/java/org/apache/nifi/processors/stateless/ExecuteStateless.java

```diff
@@ -0,0 +1,1202 @@
+        final File contentRepoDirectory;
+        if (CONTENT_STORAGE_DISK.getValue().equals(contentStorageStrategy)) {
+            final File contentRepoRootDirectory = new File(workingDirectory, "flowfile-content");
+            final File processorContentRepo = new File(contentRepoRootDirectory, getIdentifier());
+            contentRepoDirectory = new File(processorContentRepo, String.valueOf(contentRepoIndex));
+        } else {
+            contentRepoDirectory = null;
+        }
+
+        return new StatelessEngineConfiguration() {
+            @Override
+            public File getWorkingDirectory() {
+                return workingDirectory;
+            }
+
+            @Override
+            public File getNarDirectory() {
+                return narDirectory;
+            }
+
+            @Override
+            public File getExtensionsDirectory() {
+                return narDirectory;
+            }
+
+            @Override
+            public File getKrb5File() {
+                return krb5Conf;
+            }
+
+            @Override
+            public Optional<File> getContentRepositoryDirectory() {
+                return Optional.ofNullable(contentRepoDirectory);
+            }
+
+            @Override
+            public SslContextDefinition getSslContext() {
+                return sslContextDefinition;
+            }
+
+            @Override
+            public String getSensitivePropsKey() {
+                return getIdentifier();
+            }
+
+            @Override
+            public List<ExtensionClientDefinition> getExtensionClients() {
+                return Collections.emptyList();
+            }
+
+            @Override
+            public boolean isLogExtensionDiscovery() {
+                return false;
+            }
+        };
+    }
+
+    @Override
+    public Collection<SearchResult> search(final SearchContext context) {
+        if (flowSnapshot == null) {
+            return Collections.emptyList();
+        }
+
+        final VersionedComponentSearchResults results = new VersionedComponentSearchResults(context.getSearchTerm());
+        final Bucket bucket = flowSnapshot.getBucket();
+        if (bucket != null) {
+            results.add(bucket.getIdentifier(), "Bucket ID");
+            results.add(bucket.getName(), "Bucket Name");
+            results.add(bucket.getDescription(), "Bucket Description");
+        }
+
+        final VersionedFlow versionedFlow = flowSnapshot.getFlow();
+        if (versionedFlow != null) {
+            results.add(versionedFlow.getIdentifier(), "Flow ID");
+            results.add(versionedFlow.getName(), "Flow Name");
+            results.add(versionedFlow.getDescription(), "Flow Description");
+        }
+
+        search(flowSnapshot.getFlowContents(), results);
+        return results.toList();
+    }
+
+    private void search(final VersionedProcessGroup group, final VersionedComponentSearchResults results) {
```

Comment:
Yeah while testing & debugging, I saw things in the logs indicating Processor ID, etc. and I couldn't search based on that. So this will be helpful when seeing something in the logs and wanting to see how that processor is used in the flow.